### ISSUE TITLE
Version 3.47.909 did a complete chat and groups data cleansing

### ISSUE THREAD

**Issue by RolandXa** [wireapp/wire-android#2803]
Today I got version 3.47.909 via Google Play Store update on different Android devices. On all of them I can't see any chats not goups. Beside the address list this looks like a total total wipe out of all communication history data. This how it looks via GUI. I did not have the chance the chaeck whether the data is really gone ..

This appeard to different Android Version (updated list)
Huawei  P20: 9.0
Samsung Tab 8:  7.0
Samsung Tab S3: 9.0 
Xiaomi Mi Note 10: 9.0
Sony XZ premium (G8141): 9.0

I'm not amused.

**Comment by penCsharpener** [comment]
I had the same happening! But the data is kind a still there but the contact list is gone. When I search my contacts the app want to start a group chat between me and them and then I can also see some chat history. 
Please revert whatever changes you made to restore the previous functionality. Android app is completely unusable.

**Comment by makingthematrix** [comment]
Please try Settings -> Advanced -> Perform full synchronization

**Comment by RolandXa** [comment]
about "Please try Settings -> Advanced -> Perform full synchronization"
Yes this works on all my Android devices. With a 'but:' The conversition I did on a iOS or Windows device, since this update and now, the moment I did this 'Perfom full sync', is gone/not visible on my Android devices. 
I fear this is a gap which can't be recovered, right?
I mean this is not a huge issue, but still, something which can be avoided with a stop of the current Play Store version and a rollout of a fixed version ASAP.

Anyway: Big thank you to you, makingthematrix, for the simple but helpful tip!

**Comment by penCsharpener** [comment]
I performed the sync as well and it helped a bit. But the damage is already done. Before the sync it seems that Wire Android considered every private conversation as a group. When I was search my contacts for those I wanted to chat with it started a group with them by itself. 
Best example: I search for my wife, clicked her contact, started a group and because that wasn't what I wanted left the group. Now after the full sync I see our conversation again but I can't write her anymore because I'm out of the 'group'. In the meantime on Wire Desktop (Win10) everything seems to work fine and I can write her and there is no mention of person added to group or person left. But on Android I see those things.

**Comment by penCsharpener** [comment]
besides after full sync all the conversations and messages were marked unread and are completely unordered. Before at least the most recent conversations showed first in the list.

**Comment by shadow00** [comment]
Same thing here, my group chat was completely gone. After a full sync the group chat shows up again, but all the messages in it are still gone.

**Comment by penCsharpener** [comment]
Ironically I can share something with a contact I have 'left the group' from. But I still can't write. The keyboard doesn't show up. This really leaves me worried every time I see a Wire update...

**Comment by Takkat-Nebuk** [comment]
This is to confirm that a full sync restored most of my chats including messages, though maqny but not all were marked as unread. Sadly one entire group including all messages is missing from the Android App. This group was then also deleted from the Linux desktop app.

Because that means **data loss** I consider this bug as severe.

**Comment by madranet** [comment]
I did a second 'Perform full synchronisation' late last night and all my individual contact conversations have been restored to their former glory [as in I can now access the keyboard to respond]. Although I suspect Wire is still confused as to whether they are Contacts or Groups as, on each one, I got a _"You added \<contact name> to the conversation"_ message, as if I was adding that person to a group.

Also, the actual group conversation which lost all of its messages is still empty, even after this second full synchronisation.

Non-tecchy family member on my contacts list also asked me yesterday [while the Wire update was busy messing things up] why she'd received messages telling her I'd added her to a group and then that I'd left it again.

![Wire 2020-04-29 at 13 31 59 (2)](https://user-images.githubusercontent.com/669658/80696938-1813c180-8ad0-11ea-85ef-1a6cf6f9b758.png)

**Comment by penCsharpener** [comment]
Same here. A second sync recovered some broken conversations 👍 but reset the order of conversations and marked everything unread again 👎 . That doesn't change that the latest update made a mess on many peoples devices. Most people are not tech-savvy. Neither would they search Github for clues. I wish the play store had a way of marking updates as faulty/buggy to warn other users.

**Comment by madranet** [comment]
> ...I wish the play store had a way of marking updates as faulty/buggy to warn other users.

It's also a bit concerning that [as of when I checked this morning] there's been no official response from Wire about this very damaging bug and  neither has  the 3,47,909 update been pulled from the Play Store. So, presumably, the amount of people affected by this is only going to increase as the update percolates across more devices.

**Comment by shadow00** [comment]
Apparently there was a commit/merge possibly related to this issue #2800, but I suspect it actually didn't fix the problem.
(Or maybe it's just something else entirely)

**Comment by madranet** [comment]
>...I don't see how it can be connected to our changes, but the fix is easy: perform full sync.

If that's the 'official' response and does indeed cover this issue, then it's pretty complacent.  

_"...don't see how it can be connected to our changes..."_ when it occurred directly after an upgrade. 

And _"...the fix is easy: perform full sync..."_ when, as we've all been pointing out, a full sync only partially restores things to how they were.

**Comment by lapofran** [comment]
I also lost all the groups and conversations with the update. A 'Perform full sync' restored the groups, but not the conversation content. This is disappointing since I already have to work extra hard to convince other people to use Wire as a secure app rather than other unsecure alternatives.

**Comment by alexleonard** [comment]
This is a bit of a mess TBH - numerous people I had convinced to sign up were already tentative and really only sticking with it because of my continued recommendation, but whilst it's great that we can recover the groups/chats, in some ways the damage is already done - and the fact that after sync it reappears all jumbled up really doesn't help win any brownie points with my contacts 😢 I suspect I'm going to lose a few more people due to this issue.

**Comment by madranet** [comment]
I'm in a similar position. After doing my research into the various options available, I spent a long time convincing all of my family circle and several friends to settle on Wire as a common secure communication platform we could all use across our various Macs, PCs, iPhones, Androids, etc.

Quite a few times since then, I've been left wondering if I did the right thing:

Wire's dev team seem incredibly reluctant to fix any of the long-standing usability issues [Notifications have been broken since forever. Ditto landscape support or even something as basic as being able to adjust font size. Nor has the supposed ability to restore from backup ever worked reliably for me]. 

But this is the first time that Wire has actually wiped out my existing data with no action on my part. I am starting to think this latest debacle may be the straw that broke the camel's back.

**Comment by agates** [comment]
I am having the same problem.

The fact of the matter is, they have to support their paying clients first.  It's annoying but sometimes I wonder if they actually use their own app?

**Comment by cocoon** [comment]
I had the same problem and additionally my messages cannot be decrypted on other devices!
I tried multiple times on different dvices to reset the session, no change.

I made a backup and tried to clear chache, data and multiple times to reinstall and restore the backup. I am at a point where I only see the people from the backup, but no messages, a fullsync is not working. It was somewhen at the beginning, but then my messages could not be decrypted so I tried ti reinstall.
It even asked after a start that it is a new device again.

**Comment by madranet** [comment]
@cocoon I've had similar problems with restoring from backup in the recent past. But they predate this latest snafu with the 3,47,909 update. So I think they're a separate issue and not directly caused by this particular update.

**Comment by stibu81** [comment]
I also had this problem. For two of my contacts, I unintentionally created group chats by clicking on them in the contact list. Unfortunately, I left on of those groups. After performing a full sync I ended up with the same situation as @madranet: My chats were back and I could leave the obsolete group chat I was still in. But to the person, where I had left the group chat, I could no longer write messages. Fortunately, after logging out and in again on my Android device, I can again write messages to that person as well.

So, in summary, full sync and log out brought back everything back to normal in my case.

**Comment by cocoon** [comment]
@madranet Yes thanks, I found not that the mentioned db migration sounds related: https://github.com/wireapp/wire-android/issues/2756

Something is broken, so it sounds like the only hope might be a future update that allows cross platform export / import or a fix for backup imports.

**Comment by makingthematrix** [comment]
Hi,

We have just released a hotfix which should force-refresh conversation lists. If you already did the full sync there should be no change for you.

**Comment by shadow00** [comment]
@makingthematrix 
> If you already did the full sync there should be no change for you.

Does that mean that if we already did a full sync we won't be able to recover group chats anymore?
I still have all the conversations on a second device (which I did not update), but on my main phone the group chat looks like I just joined in, no old messages show up.
Is there no way to perhaps copy old data from another phone? Or is it just gone?

**Comment by Porkepix** [comment]
After update, some private conversations had an unread messages count notification for old messages read from a long time ago (2019), I guess this is related?
One or two other conversations got this notification appearing when going to read it, and disappeared once opened and back to conversation list. I also guess that's related?

**Comment by makingthematrix** [comment]
@shadow00 :
 We are investigating different scenarios and trying to recover from this issue due to a big migration we are doing. Sorry for the inconvenience. Please send us your logs. That would help us to have more context and address the problem in a better way.

@Porkepix : Changes in notifications appearing are not related to this.

**Comment by Porkepix** [comment]
> @Porkepix : Changes in notifications appearing are not related to this.

Because I might have not been clear (English isn't my native language), I wasn't talking about Android notifications, but the number appearing besides contacts/conversations showing the number of unread messages. It didn't do that previously, and as this issue seems to be about messed up history, it seemed to maybe have been related in some ways.

It also obviously can be completely unrelated.

**Comment by makingthematrix** [comment]
@Porkepix : The unread messages number will disappear after you scroll down to the newest message.

**Comment by Porkepix** [comment]
> @Porkepix : The unread messages number will disappear after you scroll down to the newest message.

Yes, but my point was that it were very old messages obviously already read :) (Like six month old messages from 2019).

**Comment by makingthematrix** [comment]
@Porkepix : Sorry. It can be due to a number of things. Recently we make a lot of changes with each release.

**Comment by IronJunkie** [comment]
Have same issue with contacts duplicated as groups, and one contact I can't type msgs to after I "left"  group as an attempt to delete duplicate contact that was shown as a group, will try log out & in again fix later. 

However anyone else also have a group icon labelled "default" with a sort of ghost Icon & only group member is yourself? Is it safe to delete or will I delete myself or something?

**Comment by makingthematrix** [comment]
@IronJunkie : It looks like some recent changes brought back to the main conversation list conversations previously marked as removed. "default" means that we no longer have the name of the conversation or the user which should be displayed at this place. You can safely delete them.

**Comment by Porkepix** [comment]
> @IronJunkie : It looks like some recent changes brought back to the main conversation list conversations previously marked as removed. "default" means that we no longer have the name of the conversation or the user which should be displayed at this place. You can safely delete them.

Does that mean deleted content… isn't really deleted…?

**Comment by makingthematrix** [comment]
@Porkepix : No, it doesn't mean that. For technical reasons some entries (but not all data) can't be deleted right away. I'd rather not go into details here :)

Ok, I guess we can close this topic now?

**Comment by maltfield** [comment]
related: As of last month, Riot now uses e2ee as the default

 * https://blog.riot.im/e2e-encryption-by-default-cross-signing-is-here/
